management-service/src/main/java/pl/pwr/zpi/notifications/discord/service/DiscordReceiverService.java

```diff
@@ -18,35 +22,83 @@ public class DiscordReceiverService {
     private final DiscordRepository discordRepository;
     private final ConfidentialTextEncoder confidentialTextEncoder;
 
+    private final String WEBHOOK_URL_REGEX = "https://discord.com/api/webhooks/[0-9]+/[a-zA-Z0-9\\-]+";
+
     public List<DiscordReceiver> getAllDiscordIntegrations() {
         return discordRepository.findAll();
     }
 
-    public void addNewDiscordIntegration(DiscordReceiverDTO discordIntegration) throws Exception {
+    public void deleteDiscordReceiver(Long receiverId) {
+        discordRepository.deleteById(receiverId);
+    }
+
+    public void createDiscordReceiver(DiscordReceiverDTO discordIntegration) throws Exception {
+        long now = System.currentTimeMillis();
+
         String encryptedWebhookUrl = confidentialTextEncoder.encrypt(discordIntegration.getWebhookUrl());
         checkIfWebhookExists(encryptedWebhookUrl);
         DiscordReceiver receiver = DiscordReceiver.builder()
                 .receiverName(discordIntegration.getName())
                 .webhookUrl(encryptedWebhookUrl)
-                .createdAt(System.currentTimeMillis())
+                .createdAt(now)
+                .updatedAt(now)
                 .build();
         discordRepository.save(receiver);
     }
 
-    public DiscordReceiver updateDiscordIntegration(Long id, DiscordReceiverDTO discordReceiver) throws Exception {
+    public DiscordReceiver updateDiscordIntegration(Long id, UpdateDiscordReceiverRequest updateDiscordReceiverRequest) throws Exception {
         var receiver = getDiscordReceiver(id);
-        String encryptedWebhookUrl = confidentialTextEncoder.encrypt(discordReceiver.getWebhookUrl());
+        String encryptedWebhookUrl = confidentialTextEncoder.encrypt(updateDiscordReceiverRequest.webhookUrl());
         checkIfUserCanUpdateWebhookUrl(encryptedWebhookUrl, id);
 
-        receiver.setReceiverName(discordReceiver.getName());
-        receiver.setWebhookUrl(encryptedWebhookUrl);
+        patchReceiver(receiver, updateDiscordReceiverRequest);
+
         receiver.setUpdatedAt(System.currentTimeMillis());
         return discordRepository.save(receiver);
     }
 
+    private void patchReceiver(DiscordReceiver discordReceiver, UpdateDiscordReceiverRequest updateRequest) {
+        if(updateRequest.name() != null) {
+            validateReceiverName(updateRequest.name());
+            discordReceiver.setReceiverName(updateRequest.name());
+        }
+
+        if(updateRequest.webhookUrl() != null) {
+            validateWebhookUrl(updateRequest.webhookUrl());
+            discordReceiver.setWebhookUrl(updateRequest.webhookUrl());
+        }
+    }
+
+    private void validateReceiverName(String name) {
+        if(name.isEmpty() || name.length() > 100) {
+           throw new RuntimeException("Receiver name length has to be > 1 and < 100");
+        }
+    }
```

Comment:
why this validation can't be done with annotation in UpdateDiscordReceiverRequest?

Reply:
The point of UpdateDiscordReceiverRequest is to make both name and webhookUrl optional fields, with validation in place they can't be optional (at least I am not aware of a way in which they can)